.github/workflows/matrix-update.yml

```diff
@@ -0,0 +1,69 @@
+name: Compatibility matrix update
+on:
+  workflow_call: {}
+
+env:
+  NODE_VERSION: "20.11.0"
+  PNPM_VERSION: "8.15.1"
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v2.2.4
+        with:
+          version: ${{ env.PNPM_VERSION }}
+
+      - name: Setup Node.js v18
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Install pnpm Dependencies
+        run: pnpm install --frozen-lockfile --filter ./tools/compatibility-matrix-automation
+
+      - name: Download Output Files
+        uses: actions/download-artifact@v3
+        with:
+          name: out
+          path: out
+
+      - name: Run Update Script
+        run: |
+          cd tools/compatibility-matrix-automation
+          OUT_FOLDER=../../out npx update-matrix
+          rm -r ../../out
```

Comment:
Is this deleted because we don't want to include them in the following commit? https://github.com/peter-evans/create-pull-request has an option to specify paths to commit instead (`add-paths`)